dashboard/lib/tasks/asset_sync.rake

```diff
@@ -0,0 +1,11 @@
+namespace :assets do
+  desc 'Synchronize assets to S3'
+  task :sync => :environment do
+    # AWS CLI implements an optimized `sync` utility without any Ruby SDK equivalent.
+    `aws s3 sync #{CDO.dashboard_dir}/public/assets s3://#{CDO.assets_bucket}/#{rack_env}/assets --acl public-read --cache-control 'max-age=31536000'`
+  end
+end
+
+Rake::Task['assets:precompile'].enhance do
+  Rake::Task['assets:sync'].invoke if CDO.daemon
```

Comment:
Does this mean we're only going to do this upload in multi-server environments (i.e. only production)?  It seems like we'd want test to do this too, and possibly staging and adhoc environments.  If adhocs do this, they'll collide on the `rack_env` prefix (I believe) so we'd want to give them something more specific.